Lib/test/test_importlib/test_util.py

```diff
@@ -775,5 +775,33 @@ def test_complete_multi_phase_init_module(self):
             self.run_with_own_gil(script)
 
 
+class MiscTests(unittest.TestCase):
+    def test_atomic_write_should_notice_incomplete_writes(self):
+        from importlib import _bootstrap_external
+        from test.support import os_helper
+        import _pyio
+        import os
+
+        oldwrite = os.write
+        seen_write = False
+
+        # emulate an os.write that only writes partial data
+        def write(fd, data):
+            nonlocal seen_write
+            seen_write = True
+            return oldwrite(fd, data[:100])
```

Comment:
Since this is a magic constant for the test, it might be best to put it in a variable and then use that variable below to guarantee the amount of bytes written is smaller than the number of bytes sent in to write.

Reply:
I introduced a name for the constant and use the constant to make sure that I pass in a longer bytes object at the call site.